src/Microsoft.AspNet.Mvc/MvcServices.cs

```diff
@@ -138,6 +139,9 @@ public static IEnumerable<IServiceDescriptor> GetDefaultServices(IConfiguration
             // Only want one ITagHelperActivator so it can cache Type activation information. Types won't conflict.
             yield return describe.Singleton<ITagHelperActivator, DefaultTagHelperActivator>();
 
+            // Consumed by the Cache tag helper to cache results across the lifetime of the application.
+            yield return describe.Singleton<IMemoryCache, MemoryCache>();
```

Comment:
I don't think we should register IMemoryCache for the rest of the application. This is just the MVC cache right?